decorator.js

```diff
@@ -21,11 +21,11 @@ function addClickOutsideListener(Component) {
     },
 
     render: function render() {
-      return React.createElement(Component, objectAssign({
+      return React.createElement(Component, objectAssign({}, this.props, {
```

Comment:
Ah, yes. I agree fixing the naming conflict (as a breaking change in a major version) is probably the best path. To expose the enable/disable functions to the wrapped component would still require merging those properties with the passed in props, but there wouldn't be a naming conflict, so order wouldn't matter unless the user passed in props which have the same names as those functions.

Can this pull be merged as a fix for now?